### PR TITLE
Return `ClientResponse` from `_request` and add methods for sensor value and unit

### PR DESCRIPTION
**Pull request by engrbm87**
## Breaking Changes
`export_sensor` has been removed. Use `get_sensor_value` and `get_sensor_unit` instead.
## Proposed Changes

- Return `ClientResponse` from `_request` to be handled by each method as required.
- Initialize `self.status_data` and `self.sensor_data` as `dict` and remove checks for `None`.
- Add separate methods to retrieve sensor value and unit.

### CODE REVIEW

**Review by engrbm87 — COMMENTED** [review]
I am not sure if we should keep this method or remove it.

File: pydroid_ipcam/__init__.py [hunk 1]
```diff
@@ -122,23 +106,17 @@ def current_settings(self) -> Dict[str, Any]:
     @property
     def enabled_sensors(self) -> List[str]:
         """Return the enabled sensors."""
-        if self.sensor_data is None:
-            return []
         return list(self.sensor_data.keys())
```
Comment (MartinHjelmare) [inline pydroid_ipcam/__init__.py:109]:
```suggestion
        return list(self.sensor_data)
```

File: pydroid_ipcam/__init__.py [hunk 2]
```diff
@@ -122,23 +106,17 @@ def current_settings(self) -> Dict[str, Any]:
...
 
     @property
     def enabled_settings(self) -> List[str]:
         """Return the enabled settings."""
-        if self.status_data is None:
-            return []
         return list(self.status_data.get("curvals", {}).keys())
```
Comment (MartinHjelmare) [inline pydroid_ipcam/__init__.py:114]:
```suggestion
        return list(self.status_data.get("curvals", {}))
```

**Comment by MartinHjelmare** [comment]
Let's remove `export_sensor`. Next release will be breaking anyway.

File: pydroid_ipcam/__init__.py [hunk 3]
```diff
@@ -177,24 +153,28 @@ async def change_setting(self, key: str, val: Union[str, int, bool]) -> bool:
             payload = "on" if val else "off"
         else:
             payload = val
-        return cast(bool, await self._request(f"/settings/{key}?set={payload}"))
+        response = await self._request(f"/settings/{key}?set={payload}")
+        return (await response.text()).find("Ok") != -1
```
Comment (MartinHjelmare) [inline pydroid_ipcam/__init__.py:157]:
```suggestion
        return cast(bool, "Ok" in (await response.text()))
```

File: pydroid_ipcam/__init__.py [hunk 4]
```diff
@@ -177,24 +153,28 @@ async def change_setting(self, key: str, val: Union[str, int, bool]) -> bool:
...
 
     async def torch(self, activate: bool = True) -> bool:
         """Enable/disable the torch."""
         path = "/enabletorch" if activate else "/disabletorch"
-        return cast(bool, await self._request(path))
+        response = await self._request(path)
+        return (await response.text()).find("Ok") != -1
```
Comment (MartinHjelmare) [inline pydroid_ipcam/__init__.py:163]:
```suggestion
        return cast(bool, "Ok" in (await response.text()))
```

File: pydroid_ipcam/__init__.py [hunk 5]
```diff
@@ -177,24 +153,28 @@ async def change_setting(self, key: str, val: Union[str, int, bool]) -> bool:
...
 
     async def focus(self, activate: bool = True) -> bool:
         """Enable/disable camera focus."""
         path = "/focus" if activate else "/nofocus"
-        return cast(bool, await self._request(path))
+        response = await self._request(path)
+        return (await response.text()).find("Ok") != -1
```
Comment (MartinHjelmare) [inline pydroid_ipcam/__init__.py:169]:
```suggestion
        return cast(bool, "Ok" in (await response.text()))
```

File: pydroid_ipcam/__init__.py [hunk 6]
```diff
@@ -177,24 +153,28 @@ async def change_setting(self, key: str, val: Union[str, int, bool]) -> bool:
...
 
     async def record(self, record: bool = True, tag: Optional[str] = None) -> bool:
         """Enable/disable recording."""
         path = "/startvideo?force=1" if record else "/stopvideo?force=1"
         if record and tag is not None:
             path = f"/startvideo?force=1&tag={URL(tag).raw_path}"
-        return cast(bool, await self._request(path))
+        response = await self._request(path)
+        return (await response.text()).find("Ok") != -1
```
Comment (MartinHjelmare) [inline pydroid_ipcam/__init__.py:177]:
```suggestion
        return cast(bool, "Ok" in (await response.text()))
```

File: pydroid_ipcam/__init__.py [hunk 7]
```diff
@@ -228,7 +208,8 @@ async def set_orientation(self, orientation: str = "landscape") -> bool:
 
     async def set_zoom(self, zoom: int) -> bool:
         """Set the zoom level."""
-        return cast(bool, await self._request(f"/settings/ptz?zoom={zoom}"))
+        response = await self._request(f"/settings/ptz?zoom={zoom}")
+        return (await response.text()).find("Ok") != -1
```
Comment (MartinHjelmare) [inline pydroid_ipcam/__init__.py:212]:
```suggestion
        return cast(bool, "Ok" in (await response.text()))
```

**Review by MartinHjelmare — APPROVED** [review]
Thanks!

Are you happy too?

**Comment by engrbm87** [comment]
> Thanks!
> 
> Are you happy too?

Looks good. 👍

**Pull request merged**